enableLeftTop.patch

Zoom-Cropの際に、sx,syを評価するようにするパッチ

$Id: enableLeftTop.patch,v 1.4 2008-05-02 17:23:20 hsur Exp $

/*
  * Copyright (C) 2006 CLES. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
  * as published by the Free Software Foundation; either version 2
  * of the License, or (at your option) any later version.
  * 
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
  * 
  * In addition, as a special exception, cles( http://blog.cles.jp/np_cles ) gives
  * permission to link the code of this program with those files in the PEAR
  * library that are licensed under the PHP License (or with modified versions
  * of those files that use the same license as those files), and distribute
  * linked combinations including the two. You must obey the GNU General Public
  * License in all respects for all of the code used other than those files in
  * the PEAR library that are licensed under the PHP License. If you modify
  * this file, you may extend this exception to your version of the file,
  * but you are not obligated to do so. If you do not wish to do so, delete
  * this exception statement from your version.
*/

--- phpthumb.functions.php.orig	2007-01-30 17:12:14.901206122 +0900
+++ phpthumb.functions.php	2007-01-30 17:12:23.220804722 +0900
@@ -416,7 +416,9 @@
 			switch ($execfunction) {
 				case 'passthru':
 					ob_start();
-					$execfunction($command);
+					// modified by cles
+					//$execfunction($command);
+					@$execfunction($command);
 					$returnvalue = ob_get_contents();
 					ob_end_clean();
 					break;
@@ -426,7 +428,9 @@
 				case 'exec':
 				default:
 					ob_start();
-					$returnvalue = $execfunction($command);
+					// modified by cles
+					//$returnvalue = $execfunction($command);
+					$returnvalue = @$execfunction($command);
 					ob_end_clean();
 					break;
 			}
--- phpthumb.class.php.org	2007-05-19 15:30:16.720948848 +0900
+++ phpthumb.class.php	2007-05-19 15:31:49.196336184 +0900
@@ -1307,6 +1307,11 @@
 //print_r($getimagesize);
 //echo '</pre>';
 
+							// added by cles
+							if( $this->zc == 2 )
+							$commandline .= ' -gravity NorthWest';
+							else
+
 							$commandline .= ' -gravity center';
 
 							if (($wAll > 0) && ($hAll > 0)) {
@@ -2543,12 +2548,16 @@
 				$this->thumbnailCropW = round($allowable_width);
 				$this->thumbnailCropX = round(($this->source_width - $allowable_width) / 2);
 
+				// added by cles
+				if( $this->zc == 2 ) $this->thumbnailCropX = 0;
 			} elseif ($scaling_Y > $scaling_X) {
 				// some of the height will need to be cropped
 				$allowable_height = $this->source_height / $scaling_Y * $scaling_X;
 				$this->thumbnailCropH = round($allowable_height);
 				$this->thumbnailCropY = round(($this->source_height - $allowable_height) / 2);
 
+				// added by cles
+				if( $this->zc == 2 ) $this->thumbnailCropY = 0;
 			} else {
 				// image fits perfectly, no cropping needed
 			}
